google-cloud-bigtable/lib/google/cloud/bigtable/read_operations.rb

```diff
@@ -159,8 +159,10 @@ def read_rows keys: nil, ranges: nil, filter: nil, limit: nil, &block
           rescue *RowsReader::RETRYABLE_ERRORS => e
             rows_reader.retry_count += 1
             raise Google::Cloud::Error.from_error(e) unless rows_reader.retryable?
-            rows_limit, row_set = rows_reader.retry_options limit, row_set
-            retry
+            resumption_option = rows_reader.retry_options limit, row_set
+            rows_limit = resumption_option.rows_limit
+            row_set = resumption_option.row_set
+            retry if resumption_option.is_complete
```

Comment:
This seems backwards, wouldn't we want to retry if the read is _not_ complete?

Reply:
got catch, I renamed the variable name from should_retry to is_complete but didnt update all the logic :) fixed it now.